projects/common/src/navigation/navigation.service.ts

```diff
@@ -42,6 +43,11 @@ export class NavigationService {
   ) {
     this.event$(RoutesRecognized)
       .pipe(
+        tap(event => {
+          if (isEmpty(this.firstNavigatedUrl)) {
+            this.firstNavigatedUrl = decodeURIComponent(event.url);
+          }
+        }),
```

Comment:
please leave a comment here on why we are storing this.

Reply:
Added